### PR TITLE
Tabs HW 06.04

### PR DESCRIPTION
**Pull request by belikovanastasya**

### CODE REVIEW

File: src/Main/EditableComponent.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import PropTypes from 'prop-types';
+
+export class EditableComponent extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      inputVisible: false,
+      spanVisible: true,
```
Comment (iliatcymbal) [inline src/Main/EditableComponent.js]: you don't need both `inputVisible` and `spanVisible`, use only one

File: src/Main/EditableComponent.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+EditableComponent.propTypes = {
+  inputVisible: PropTypes.bool,
+  spanVisible: PropTypes.bool,
+  spanValue: PropTypes.string
+};
+EditableComponent.defaultProps = {
+  inputVisible: false,
+  spanVisible: true,
+  spanValue: 'Click to edit'
+};
```
Comment (iliatcymbal) [inline src/Main/EditableComponent.js]: why do you place state's fields into propTypes and defaultProps? State and props are different objects.

File: src/Main/Geo.js [hunk 3]
```diff
@@ -1,21 +1,46 @@
...
+        <img src={src} alt="" />
       </div>
     );
   }
 }
+
+
+Geo.propTypes = {
+  long: PropTypes.string,
+  lat: PropTypes.string
```
Comment (iliatcymbal) [inline src/Main/Geo.js]: again - why is the state in the propTypes? You don't have `long` and `lat` in properties.

File: src/Main/Main.js [hunk 4]
```diff
@@ -81,15 +88,38 @@ export class Main extends Component {
...
+Button.propTypes = {
+  addClass: PropTypes.bool,
+  hidden: PropTypes.bool
+};
+Main.propTypes = {
+  users: PropTypes.array,
+  posts: PropTypes.array,
+  show: PropTypes.bool
+};
+
```
Comment (iliatcymbal) [inline src/Main/Main.js]: For some reason you're using state as a props. Where did you get this?
state is defined in component `this.state = {...}`

and props are defined when you create component. i.e. <User properties... />
`<Comp prop1="one" prop2="two" />` --> this.props ==> { prop1: "one", prop2: "two" }

propTypes and defaultProps are only for props, state shouldn't be used inside type checking

File: src/Main/Timer.js [hunk 5]
```diff
@@ -21,4 +23,8 @@ export class Timer extends Component {
       </div>
     );
   }
-}
\ No newline at end of file
+}
+Timer.propTypes = {
+  time: PropTypes.string
```
Comment (iliatcymbal) [inline src/Main/Timer.js]: time is not a property

**Review by iliatcymbal — COMMENTED** [review]
Please, see my comments and make fixes. After that let me know to check again

**Comment by belikovanastasya** [comment]
One mistake through the whole task. Probable, I didn't understand the topic. I read 
 it again and fixed it. Check, please.

**Comment by iliatcymbal** [comment]
ok, good, please merge

**Pull request merged**